saw-core/src/SAWCore/SharedTerm.hs

```diff
@@ -1071,10 +1069,8 @@ ctxCtorElimType sc d_top c
   helper _d params pret prevs [] ret_ixs =
     -- If we are finished with our arguments, construct the final result type
     -- (p_ret ret_ixs (c params prevs))
-    do (vars, prev_vars) <- ctxVars2 sc (params ++ [pret]) prevs
-       -- note: length vars == length (params ++ [pret])
-       let param_terms = init vars
-       let p_ret = last vars
+    do param_terms <- traverse (scExtCns sc) params
+       ([p_ret], prev_vars) <- ctxVars2 sc [pret] prevs
```

Comment:
Irrefutably matching on `[p_ret]` like this relies on the `MonadFail IO` instance, which will give a very generic error in the event that this fails to match. I would prefer to explicitly match on the result of calling `ctxVars2` and explicitly panicking in the event that the first element of the pair is not a singleton list.

Reply:
Function `ctxVars2` always returns lists the same lengths as its argument lists, so this is approximately as safe as `[x, y, z] <- mapM foo [a, b, c]`. In any case, the follow-on PR #2466 is already slated to remove this line of code (a2d5adfc1b9657dc84c43869af140db901d724b1).